### PR TITLE
ensure connect events are always sent

### LINKED ISSUE #2019: Account connection events not sent for custom authentication
## Defect

Account connect events is not sent when a custom authenticator is set:

https://github.com/nats-io/nats-server/blob/4784205c2d60ee57f1c539e3eb5f19d03974a71d/server/auth.go#L323-L334

Above the events are only sent in `processClientOrLeafAuthentication`

### PR DESCRIPTION
**Pull request by ripienaar**
Fixes #2019 

Signed-off-by: R.I.Pienaar <rip@devco.net>

/cc @nats-io/core

### CODE REVIEW

**Review by derekcollison — CHANGES_REQUESTED** [review]

File: server/auth.go [hunk 1]
```diff
@@ -327,10 +327,21 @@ func (s *Server) isClientAuthorized(c *client) bool {
 	// multiple users with TLS map if enabled, then token,
 	// then single user/pass.
 	if opts.CustomClientAuthentication != nil {
-		return opts.CustomClientAuthentication.Check(c)
+		if !opts.CustomClientAuthentication.Check(c) {
```
[the remark this answers is not in the document]
Comment (ripienaar) [reply server/auth.go]: done

File: .travis.yml [hunk 2]
```diff
@@ -4,7 +4,6 @@ dist: focal
 language: go
 go:
 - 1.16.x
-- 1.15.x
```
Comment (derekcollison) [inline .travis.yml:7]: I think we do different tests based on 1.15 for now so might want to revert this one.

Comment (ripienaar) [reply .travis.yml:7]: Looks like just 

https://github.com/nats-io/nats-server/blob/4784205c2d60ee57f1c539e3eb5f19d03974a71d/.travis.yml#L32

I can update that too? Already go.mod declares 1.16 required so worth sorting it out

Comment (derekcollison) [reply .travis.yml:7]: We run different tests on each platform so we always have 2... Let's leave for now IMO. Agree be nice to remove but will want Ivan to weigh in.

Comment (ripienaar) [reply .travis.yml:7]: ok will revert

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**